plugins/aws/resoto_plugin_aws/resource/base.py

```diff
@@ -119,6 +120,24 @@ def to_json(self) -> Json:
             ),
         )
 
+    def set_arn(
+        self,
+        builder: GraphBuilder,
+        region: Optional[str] = None,
+        service: Optional[str] = None,
+        account: Optional[str] = None,
+        resource: str = "",
+    ) -> None:
+        aws_region = builder.region
+        partition = arn_partition(aws_region)
+        if region is None:
+            region = aws_region.id
```

Comment:
wouldn't we need to handle the global region different (like empty)?

Reply:
In that case you put `region=""`, same for `account` which isn't always filled. See [this arn for apigateway for example](https://github.com/someengineering/resoto/blob/3ed3f3be24c4b212f028b4f5c47b4dc401438afd/plugins/aws/test/resources/apigateway_test.py#L13)